### DISCUSSION TITLE
SQL server support

### DISCUSSION THREAD

**Discussion by VenkateshSrini** [JasperFx/marten#1444] [Ideas] [answered]
Hi,

This is a more a feature that I ask. Is there a plan for marten to support SQL server?

**Comment by oskardudycz** [comment]
@VenkateshSrini I assume that you mean MSSQL server? We made a survey some time ago about adding support for the SQL Server for our users (or other DB engines), but it appeared that most of the users that would like to use Marten don't mind to use the Postgres. Adding new DB Engines is a significant amount of time. 

Currently, we don't have enough capacity. For the next release (v4) we'll be more focused on the Event Store part. 

We have plans to do some substantial changes in v4 to make our codebase more modular and in the future support other DB engines, but that won't happen soon.

What features would you find the most important for you? 

Are your project MS vendor locked in? Would you consider using Postgres?

**Comment by VenkateshSrini** [comment]
Yes the client is almost having big SQL server database established over years. Also getting new data base into their db cluster is night mare for them as they want to move there data to Azure SQL or cosmos.

**Comment by oskardudycz** [comment]
Understood. So we hope that we'll add the MSSQL server support at some point, but unfortunately, that won't happen in the nearest future.

**Comment by jeremydmiller** [comment]
@VenkateshSrini Sql Server still lags Postgresql in JSON support, but at least they're moving in the right direction. I think the adoption of sql/json will make a big difference for Marten's ability to support other database engines in the future.

**Comment by jeremydmiller** [comment, marked as answer]
Closing because there's no real resolution any time soon.
